sql/ddl_api.sql

```diff
@@ -216,3 +216,6 @@ CREATE OR REPLACE PROCEDURE @extschema@.refresh_continuous_aggregate(
     force                    BOOLEAN = FALSE
 ) LANGUAGE C AS '@MODULE_PATHNAME@', 'ts_continuous_agg_refresh';
 
+CREATE OR REPLACE PROCEDURE @extschema@.move_hypertable_invalidations(
```

Comment:
I suggest putting this function in _timescaledb_functions schema. I don't see how this functionality deserves presence in public/extension schema at this point.

Reply:
Actually, why does the move function work on only a specific hypertable? Shouldn't it do the work for _all_ hypertables?

At least I think it should be an option to process the entire log in one go. This makes more sense if we expect this to be done by an async job and not a refresh on a specific hypertable.

Reply:
The function `move_invalidations_from_hyper_to_cagg_log` processes this is the wrong order: for each continuous aggregate, process all hypertable invalidations and incorporate those relevant for the continuous aggregate.

Processing the hypertable invalidation log in one go and collect the invalidations instead is a sensible change, but it is going to be bigger so will take that as a separate pull request.

It is straightforward to add overloads for supporting this once the refactoring is done.